ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue

```diff
@@ -0,0 +1,133 @@
+      </v-row>
+
+      <v-row>
+        <v-col cols="2">
+          <label class="generic-label ml" for="">Year of Manufacture</label>
+        </v-col>
+        <v-col cols="4">
+          <v-text-field
+            v-model="yearOfManufacture"
+            filled
+            label="Year of Manufacture"
+            persistent-hint
+            hint="YYYY"
+          />
+        </v-col>
+        <v-col cols="6">
+          <v-checkbox>
+            <template v-slot:label>
+              This Year of Manufacture is approximate
+            </template>
+          </v-checkbox>
+        </v-col>
+      </v-row>
+
+      <v-divider class="ma-0 pa-0 mt-2 mb-5" />
+
+      <v-row>
+        <v-col cols="2">
+          <label class="generic-label ml" for="">Make</label>
+        </v-col>
+        <v-col cols="10">
+          <v-text-field v-model="make" filled label="Make" />
+        </v-col>
+      </v-row>
+
+      <v-row>
+        <v-col cols="2">
+          <label class="generic-label ml" for="">Model</label>
+        </v-col>
+        <v-col cols="10">
+          <v-text-field v-model="model" filled label="Model" />
+        </v-col>
+      </v-row>
+    </v-card>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, reactive, toRefs, watch } from '@vue/composition-api'
+import { useGetters, useActions } from 'vuex-composition-helpers'
+
+export default defineComponent({
+  setup () {
+    const {
+      getMhrRegistrationManufacturerName,
+      getMhrRegistrationYearOfManufacture,
+      getMhrRegistrationHomeMake,
+      getMhrRegistrationHomeModel
+    } = useGetters<any>([
+      'getMhrRegistrationManufacturerName',
+      'getMhrRegistrationYearOfManufacture',
+      'getMhrRegistrationHomeMake',
+      'getMhrRegistrationHomeModel'
+    ])
+    const {
+      setMhrRegistrationManufacturerName,
+      setMhrRegistrationYearOfManufacture,
+      setMhrRegistrationHomeMake,
+      setMhrRegistrationHomeModel
+    } = useActions<any>([
+      'setMhrRegistrationManufacturerName',
+      'setMhrRegistrationYearOfManufacture',
+      'setMhrRegistrationHomeMake',
+      'setMhrRegistrationHomeModel'
+    ])
+
+    const localState = reactive({
+      manufacturerName: getMhrRegistrationManufacturerName.value,
+      yearOfManufacture: getMhrRegistrationYearOfManufacture.value,
+      make: getMhrRegistrationHomeMake.value,
+      model: getMhrRegistrationHomeModel.value
+    })
+
+    watch(
+      () => localState.manufacturerName,
+      (val: string) => {
+        setMhrRegistrationManufacturerName(val)
+      }
+    )
+
+    watch(
+      () => localState.yearOfManufacture,
+      (val: string) => {
+        setMhrRegistrationYearOfManufacture(val)
+      }
+    )
+
+    watch(
+      () => localState.make,
+      (val: string) => {
+        setMhrRegistrationHomeMake(val)
+      }
+    )
+
+    watch(
+      () => localState.model,
+      (val: string) => {
+        setMhrRegistrationHomeModel(val)
+      }
+    )
```

Comment:
It's a cool way to do this, seeing how it's not a form with standard submission. 

That being said, correct me if i'm wrong, but this will apply the values to store immediately regardless of validations no?
Is there field level validations or is the plane to intercept here prior to setting properties to store if `invalid`?

Reply:
Right now, we are storing values. We can either validate them prior or use stored values for validation. We also discussed in one of the meetings that they wanted to save a draft with (possibly) invalid values.

Reply:
Will leave this comment unresolved but the assumption is that validations will be applied when you continue work on this component.